gapis/api/vulkan/api/command_buffer_control.api

```diff
@@ -482,24 +482,30 @@ cmd VkResult vkBeginCommandBuffer(
   } else {
     buff := CommandBuffers[commandBuffer]
     clear(buff.ImageTransitions)
-    if (buff.Level == VK_COMMAND_BUFFER_LEVEL_SECONDARY) && (info.pInheritanceInfo != null) {
+    if (info.pInheritanceInfo != null) {
+      // Make sure to read InheritanceInfo even if the command buffer is not
+      // secondary. The spec allows a primary command buffer to have a non-null
+      // pInheritanceInfo, it will just be ignored. Not reading may lead to
+      // crashes in the validation layer at replay time, see b/161126544.
       inheritanceInfo := info.pInheritanceInfo[0]
```

Comment:
If buff.Level != VK_COMMAND_BUFFER_LEVEL_SECONDARY, the pInheritanceInfo pointer may just be junk. You *must not* read through it in that case.

If you really want to be robust you should instead force the value to null in the primary case. Then at replay time, there is no trouble.

Please pursue the validation layer fix if they are incorrectly reading through this pointer.